### PR TITLE
Update deno dependencies, use deno tasks

### PR DESCRIPTION
**Pull request by LeoDog896**
This adds 2 changes:

- Add deno-outdated to update dependencies
- Use deno tasks instead of makefile

### CODE REVIEW

**Comment by lino-levan** [comment]
@LeoDog896 can you fix this PR, this is blocking https://github.com/denoland/fresh/pull/986

**Comment by LeoDog896** [comment]
Thanks! Will do once I get home, almost forgot about this PR.

**Comment by lino-levan** [comment]
@lucacasonato could you please take a look? This is blocking updates to fresh. Thanks!

**Comment by LeoDog896** [comment]
Do note the reason why the tests are changed is because the stdlib is *still* unstable -- instead of dealing with the new crypto API, I moved to collections. If this is not preferred, let me know.

**Comment by lino-levan** [comment]
@LeoDog896 both std and esbuild got new releases. Please update this? I know it sucks that this PR isn't getting merged but ideally when it does it should be as up to date as possible.

**Comment by lino-levan** [comment]
Gently pinging @lucacasonato to please take a look!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -30,11 +30,11 @@ jobs:
 
       - name: Format
         if: runner.os == 'Linux'
-        run: make fmt/check
+        run: deno check
```
Comment (lucacasonato) [inline .github/workflows/ci.yml:33]: This is not the same. `deno fmt --check` != `deno check`

The latter is for type checking.

File: deno.jsonc [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
+  "tasks": {
+    "test": "deno test -A",
+    "lint": "deno lint",
+    "fmt": "deno fmt",
+    "check": "deno fmt --check",
```
Comment (lucacasonato) [inline deno.jsonc:6]: Remove these. This adds nothing, as it's just a pure alias.

**Review by lucacasonato — APPROVED** [review]
Thanks

**Comment by lino-levan** [comment]
LGTM! Excited for when this lands.

**Comment by lucacasonato** [comment]
Tests don't pass unfortunately

**Comment by lino-levan** [comment]
Just pinged @LeoDog896 on discord, this should fix the tests. Could you approve the workflow again @lucacasonato?

**Pull request merged**